qiskit/algorithms/evolvers/real/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,213 @@
+        return self._expectation
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        """
+        Whether computing the expectation value of auxiliary operators is supported.
+
+        Returns:
+            True if `aux_operators` expectations in the EvolutionProblem can be evaluated, False
+                otherwise.
+        """
+        return True
+
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
+        """
+        Evolves a quantum state for a given time using the Trotterization method
+        based on a product formula provided.
+
+        Note: Time-dependent Hamiltonians are not yet supported.
+
+        Args:
+            evolution_problem: Instance defining evolution problem. For the included Hamiltonian,
+                only SummedOp, PauliOp are supported by TrotterQrte.
+
+        Returns:
+            Evolution result that includes an evolved state.
+
+        Raises:
+            ValueError: If `t_param` is not set to None in the EvolutionProblem (feature not
+                currently supported).
+            ValueError: If the `initial_state` is not provided in the EvolutionProblem.
+        """
+        if evolution_problem.t_param is not None:
+            raise ValueError(
+                "TrotterQrte does not accept a time dependent hamiltonian,"
+                "`t_param` from the EvolutionProblem should be set to None."
+            )
+
+        if evolution_problem.aux_operators is not None and (
+            self._quantum_instance is None or self._expectation is None
+        ):
+            raise ValueError(
+                "aux_operators where provided for evaluations but no `expectation` or "
+                "`quantum_instance` was provided."
+            )
+
+        hamiltonian = self._try_binding_params(
+            evolution_problem.hamiltonian, evolution_problem.hamiltonian_value_dict
+        )
+        # the evolution gate
+        evolution_gate = CircuitOp(
+            PauliEvolutionGate(hamiltonian, evolution_problem.time, synthesis=self._product_formula)
+        )
+
+        if evolution_problem.initial_state is not None:
+            quantum_state = evolution_gate @ evolution_problem.initial_state
+            evolved_state = self._circuit_sampler.convert(quantum_state).eval()
+
+        else:
+            raise ValueError("`initial_state` must be provided in the EvolutionProblem.")
+
+        evaluated_aux_ops = None
+        if evolution_problem.aux_operators is not None:
+            evaluated_aux_ops = eval_observables(
+                self._quantum_instance,
+                quantum_state.primitive,
+                evolution_problem.aux_operators,
+                self._expectation,
+                algorithm_globals.numerical_tolerance_at_0,
+            )
+
+        return EvolutionResult(evolved_state.eval(), evaluated_aux_ops)
+
+    @staticmethod
+    def _try_binding_params(
```

Comment:
two brief questions on this method:
1. Doesn't `hamiltonian.bind_parameters(value_dict)` already implement parameter binding? If not, I think it should 🙂 
2. If the `EvolutionProblem.hamiltonian` can be any `OperatorBase`, shouldn't this function also accept any `OperatorBase` as input?

Reply:
1. Correct. I realized I just need to make sure to convert `SummedOp` to `PauliSumOp` after binding because this is what the PauliEvolutionGate accepts.
2. We assume that this algorithm accepts only `SummedOp`, `PauliOp`, and soon `PauliSumOp`. It is highlighted in the docs and errors thrown. The `EvolutionProblem` class is general and can be used by all algos. Not sure if specializations w.r.t typing would be worth it.